src/validation.cpp

```diff
@@ -3093,16 +3093,26 @@ void CChainState::ResetBlockFailureFlags(CBlockIndex *pindex) {
         }
         it++;
     }
+	
+	CBlockIndex *pindexFirstInvalid = nullptr;
 
     // Remove the invalidity flag from all ancestors too.
     while (pindex != nullptr) {
         if (pindex->nStatus & BLOCK_FAILED_MASK) {
+			if (pindex->nStatus & BLOCK_FAILED_VALID) {
+				pindexFirstInvalid = pindex;
+			}
```

Comment:
Changing tabs to spaces:

```suggestion
         if (pindex->nStatus & BLOCK_FAILED_VALID) {
            pindexFirstInvalid = pindex;
         }
```